### PR TITLE
Drop usage of `flatpickr` in `TimeframeForm`

### LINKED ISSUE #128: Disable autocomplete when adding new timeframe
## Describe the bug

Filling a textual datetime description for Start and End of a new Timeframe results in an autocompleted absolute date. To workaround this I added the Timeframe directly to the database.

## To Reproduce

1.  Create new Timeframe
2. Add `-30 days` for Start
3. Move curser to another input field
4. Start becomes `12/1/2021, 12:00 AM GMT+1`

## Expected behavior
The textual datetime description should remain.

## Your Environment

* Module version: 0.10.0
* Dependent module versions: 
* Icinga Web 2 version and modules (System - About): 2.10.1
* Web browser and version:
* PHP version used (`php --version`): 7.4.3
* Browser: Google Chrome Version 101.0.4951.64 (Official Build) (x86_64)

## Additional context

This may be due to a browser "feature", but perhaps this can be disabled.

### PR DESCRIPTION
**Pull request by yhabteab**
This PR removes the use of `flatpickr` and replaces it with `localdatetime`, which automatically fixes the issues mentioned below. Additionally, modals can now be used to create and edit time frames.

fixes #128, #154, #172

closes #151

### CODE REVIEW

File: library/Reporting/Web/Forms/TimeframeForm.php [hunk 1]
```diff
@@ -41,34 +40,93 @@ public function hasBeenSubmitted(): bool
...
+                    'value'       => $end,
+                    'label'       => $this->translate('End'),
+                    'description' => $this->translate('Specifies the end time of this timeframe')
+                ])
+            );
+        } else {
+            $this->addElement('text', 'end', [
+                'required'    => true,
+                'label'       => $this->translate('End'),
+                'placeholder' => $this->translate('Provide a textual datetime description'),
```
Comment (TAINCER) [inline library/Reporting/Web/Forms/TimeframeForm.php:126]: Maybe provide a usage example as the placeholder. (E.g. -30 days)

Same for the textual start time

**Review by TAINCER — COMMENTED** [review]
This seems more like a bugfix than a Refactor, since there is quite a bit of functionality added. Maybe consider adding a notice in the description.

**Comment by yhabteab** [comment]
@cla-bot check

**Review by TAINCER — APPROVED** [review]
LGTM

**Comment by yhabteab** [comment]
I have added a callback validator for the textual date-time description.

**Review by lippserd — APPROVED** [review]

**Pull request merged**